src/main/java/eu/europeana/metis/sandbox/controller/DatasetController.java

```diff
@@ -105,13 +107,24 @@ public DatasetIdDto harvestDatasetFromFile(
       @ApiParam(value = "name of the dataset", required = true) @PathVariable(value = "name") String datasetName,
       @ApiParam(value = "country of the dataset", required = true, defaultValue = "Netherlands") @RequestParam Country country,
       @ApiParam(value = "language of the dataset", required = true, defaultValue = "Dutch") @RequestParam Language language,
+      @ApiParam(value = "step size to apply in record selection") @RequestParam(required = false) Integer stepsize,
       @ApiParam(value = "dataset records uploaded in a zip file", required = true) @RequestParam MultipartFile dataset,
       @ApiParam(value = "xslt file to transform to EDM external") @RequestParam(required = false) MultipartFile xsltFile) {
     checkArgument(NAME_PATTERN.matcher(datasetName).matches(), MESSAGE_FOR_DATASET_VALID_NAME);
+    if(stepsize != null) {
+      checkArgument(stepsize > 0, MESSAGE_FOR_STEP_SIZE_VALID_VALUE);
+    }
```

Comment:
As a suggestion I saw this constant in `HarvestServiceImpl ` you can reuse it here I think or if you see a better way to place it is also good.
```
else {
  stepSize = DEFAULT_STEP_SIZE;
}
```
include it in all where the stepSize is needed as well.

Reply:
That's a nice idea, I'll do that

Reply:
Actually, changed my mind, I've decided to keep the variable in `HarvestServiceImpl`, just in case somehow there is a `null` value passed as `stepSize` the class still manages to work with the default behavior

Reply:
yes, I see.